documentation/matlab-api.md

```diff
@@ -0,0 +1,381 @@
+```
+
+If `call_MATLAB` is implemented along the lines of:
+```
+def call_MATLAB(func, *args, **kwargs):
+    call_matlab_function = print # For purposes of running this
+    args = list(args)
+    for arg in kwargs.items(): # Accumulate kwargs into args to unpack
+        args += [*arg]
+    call_matlab_function(func, *args)
+```
+It is trivial to map Python kwargs to MATLAB keyword-values
+
+
+#### Pros and Cons
+
+##### Pros
+
+- Can eventually use built-in MATLAB parser (should be kept stable).
+- More Pythonic and should lead to a consistent interface between the two.
+- Easier to standardise (single, consistent syntax)
+
+##### Cons
+
+- More typing required by user
+
+#### Transition implications
+
+- Requires Python implementation similar to that as noted above
+- Replace `parse_arguments` flag parser temporarily with variant printing deprecation warning and allow both syntaxes to coexist for a period
+- Need to check for flags in public APIs, can't check user-scripts
+
+
+# Proposal
+
+- Use positional parameters for core data
+- Use optional parameters where appropriate
+- Use key-value arguments for remaining arguments, including  rather than flag/no-flag arguments
+
+The use of `[-][no]flagname` is not supported in Python, other than as one or more "string value optional arg" and is not Pythonic. 
+The conversion for MATLAB flag-type args could be handled in the Python <-> MATLAB wrapper as there's a well defined mapping between key/value and flag syntax, but could also be avoided.
+
+# Appendix
+
+## Definitions
+
+### Flags
+Flags are a common way of distinguishing arguments from operating modes and are used in many standard programs such as POSIX compliant programs and Windows command line tools to allow extension of operation without limiting the operands of a particular function or requiring a separate program.
+
+e.g.
+```
+ls -l --sort=size *.txt
+dir /w /o:s *.txt
+```
+Here, `-l` and `/w` are flags, which set a relevant operation mode to `true`. `--sort=size` and `/o:s` are flags which set an internal variable to a particular value as described by their argument.
+If not separated by some extra marker (`-` in POSIX, `/` in Windows), these would instead by interpreted as arguments to the program.
+
+### Positional arguments
+Positional arguments are arguments which appear at a particular position in a function signature or call.
+These are usually mapped one-to-one or many-to-one from signature to call.
+e.g
+Signature:
+```
+MATLAB: function function_name(arg1, arg2, varargin)
+PYTHON: def function_name(arg1, arg2, *args)
+```
+Both would be called as
+```
+function_name(a, b, c, d, ...)
+```
+Where `a` maps to `arg1`, `b` to `arg2`, `c` and `d` to a cell array (`varargin`) in MATLAB and a tuple (`args`) in Python.
+
+### Keyword-Value Pairs
+Keyword value pairs are a common means of allowing a variable number of arguments without requiring that they be in any particular order. Typically they follow positional arguments.
```

Comment:
In the call. In the function/def line it is different

Reply:
Which call? What is different? Sorry, I don't understand.